server/cmd/usermatches/main.go

```diff
@@ -0,0 +1,257 @@
+// main things: (1) apply some filtering, and (2) write the match data out
+// somewhere, which in this app is a CSV file.
+func MarketMatchesStreaming(storage db.DEXArchivist, base, quote uint32, includeInactive bool, N int64, f func(*MatchData) error) (int, error) {
+	baseAsset := assets[base]
+	if baseAsset == nil {
+		return 0, fmt.Errorf("asset %d not found", base)
+	}
+	quoteAsset := assets[quote]
+	if quoteAsset == nil {
+		return 0, fmt.Errorf("asset %d not found", quote)
+	}
+	fDB := func(md *db.MatchDataWithCoins) error {
+		matchData := convertMatchData(baseAsset, quoteAsset, md)
+		return f(matchData)
+	}
+	return storage.MarketMatchesStreaming(base, quote, includeInactive, N, fDB)
+}
+
+func main() {
+	if err := mainCore(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+	os.Exit(0)
+}
+
+func mainCore() error {
+	ctx, quit := context.WithCancel(context.Background())
+	defer quit()
+	killChan := make(chan os.Signal, 1)
+	signal.Notify(killChan, os.Interrupt)
+	go func() {
+		<-killChan
+		quit()
+		fmt.Println("Shutting down...")
+	}()
+
+	flag.Parse()
+
+	base, quote := uint32(*base), uint32(*quote)
+	name, err := dex.MarketName(base, quote)
+	if err != nil {
+		return err
+	}
+	mkt := &dex.MarketInfo{
+		Name:  name,
+		Base:  base,
+		Quote: quote,
+	}
+
+	pgCfg := &pg.Config{
+		Host:      *dbhost,
+		Port:      strconv.Itoa(*dbport),
+		User:      *dbuser,
+		Pass:      *dbpass,
+		DBName:    *dbname,
+		MarketCfg: []*dex.MarketInfo{mkt},
+		Net:       dex.Mainnet,
+	}
+	archiver, err := pg.NewArchiverForRead(ctx, pgCfg)
+	if err != nil {
+		return err
+	}
+	defer archiver.Close()
+
+	var acctID account.AccountID
+	var haveAccount bool
+	switch len(*acct) {
+	case 0: // no account filter
+	case account.HashSize * 2:
+		acctB, err := hex.DecodeString(*acct)
+		if err != nil {
+			return err
+		}
+		copy(acctID[:], acctB)
+		haveAccount = true
+	default:
+		return fmt.Errorf("bad acct ID %v", *acct)
+	}
+
+	csvfile, err := os.Create(fmt.Sprintf("acct_matches_%v.csv", acctID))
+	if err != nil {
+		return fmt.Errorf("error creating csv file: %w", err)
+	}
+	defer csvfile.Close()
+
+	csvwriter := csv.NewWriter(csvfile)
+	defer csvwriter.Flush()
+
+	err = csvwriter.Write([]string{"unixtime", "maker", "taker", "quantity", "rate",
+		"isTakerSell", "makerSwapTx", "makerSwapVout", "makerRedeemTx", "makerRedeemVin",
+		"takerSwapTx", "takerSwapVout", "takerRedeemTx", "takerRedeemVin"})
+	if err != nil {
+		return fmt.Errorf("ERROR: csvwriter.Write failed: %w", err)
+	}
+
+	splitTx := func(txinout string) (tx, vinout string, err error) {
+		if txinout == "" {
+			return // ok
+		}
+		txsplit := strings.Split(txinout, ":")
+		if len(txsplit) != 2 {
+			err = fmt.Errorf("txinout (%s) not formatted as a txin/out", txinout)
+			return
+		}
+		_, err = strconv.ParseUint(txsplit[1], 10, 32)
+		if err != nil {
+			err = fmt.Errorf("strconv.ParseUint(%s): %w", txsplit[1], err)
+			return
+		}
+		return txsplit[0], txsplit[1], nil
+	}
```

Comment:
Ethereum is going to break this.

Reply:
Mmmm, yeah.  I guess it remains to be seen what pieces of the MatchData for eth are what.  Can cross that bridge if we want eth as a supported asset for usermatches.